src/app/loadout/Loadouts.tsx

```diff
@@ -79,15 +80,32 @@ export default function LoadoutsContainer({ account }: { account: DestinyAccount
 
 function Loadouts({ account }: { account: DestinyAccount }) {
   const location = useLocation();
+  const dispatch = useDispatch();
+
   const locationStoreId = (location.state as { storeId: string } | undefined)?.storeId;
-  const stores = useSelector(sortedStoresSelector);
-  const currentStore = getCurrentStore(stores)!;
-  const [selectedStoreId, setSelectedStoreId] = useState(
-    locationStoreId && locationStoreId !== 'vault' ? locationStoreId : currentStore.id,
+  const stores = useSelector(storesSelector);
+  const selectedStore = useSelector(selectedLoadoutStore);
+
+  const setSelectedStoreId = useCallback(
+    (storeId: string) => {
+      dispatch(updateLoadoutStore({ storeId }));
+    },
+    [dispatch],
   );
+
+  // Id the location store matches some store id, isn't the vault, and isn't the currently
+  // selected store id we need to dispatch an update to update the page.
+  if (
+    locationStoreId &&
+    stores.some((store) => store.id === locationStoreId) &&
+    locationStoreId !== 'vault' &&
+    selectedStore.id !== locationStoreId
+  ) {
+    setSelectedStoreId(locationStoreId);
+  }
```

Comment:
I haven't yet figured out how to test this but I will try again tonight. If anyone knows off the top of their head please let me know.